misc/experimental/promise/api/celix/impl/SharedPromiseState.h

```diff
@@ -208,12 +224,36 @@ namespace celix::impl {
 *********************************************************************************/
 
 template<typename T>
-inline celix::impl::SharedPromiseState<T>::SharedPromiseState(const tbb::task_arena& _executor) : executor{_executor} {}
+std::shared_ptr<celix::impl::SharedPromiseState<T>> celix::impl::SharedPromiseState<T>::create(std::shared_ptr<celix::IExecutor> _executor, int priority) {
+    auto state = std::shared_ptr<celix::impl::SharedPromiseState<T>>{new celix::impl::SharedPromiseState<T>{std::move(_executor), priority}};
+    std::weak_ptr<celix::impl::SharedPromiseState<T>> self = state;
+    state->setSelf(std::move(self));
```

Comment:
[In c++17 `state->setSelf(state);` should work. ](https://en.cppreference.com/w/cpp/language/class_template_argument_deduction) Otherwise, try `state->setSelf(std::weak_ptr(state));`.

Reply:
yes that works.